jsx/Tabs.js

```diff
@@ -141,6 +141,123 @@ Tabs.defaultProps = {
   updateURL: false
 };
 
+/**
+ * Allows to dynamically render vertical tabs corresponding to tab panes.
+ */
+
+class VerticalTabs extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const hash = window.location.hash;
+    let activeTab = "";
+
+    /**
+     * Determine the initial active tab in this order
+     * 1. Try to infer from the URL, otherwise
+     * 2. Try to infer from the defaultTab prop, otherwise
+     * 3. Set to be the first tab of the list
+     */
+    if (this.props.updateURL && hash) {
+      activeTab = hash.substr(1);
+    } else if (this.props.defaultTab) {
+      activeTab = this.props.defaultTab;
+    } else if (this.props.tabs.length > 0) {
+      activeTab = this.props.tabs[0].id;
+    }
+
+    this.state = {
+      activeTab: activeTab
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+    this.getTabs = this.getTabs.bind(this);
+    this.getTabPanes = this.getTabPanes.bind(this);
+  }
+
+  handleClick(tabId, e) {
+    this.setState({activeTab: tabId});
+    this.props.onTabChange(tabId);
+
+    // Add tab href to URL querystring and scroll the page to top
+    if (this.props.updateURL) {
+      const scrollDistance = $('body').scrollTop() || $('html').scrollTop();
```

Comment:
I don't understand what this is trying to do. Can you add a comment?

(The scrolling part and the relation between html and body in this and the line below.. the hash part is clear)

Reply:
copy of line 69 - regular tabs

I have no idea what it does, we should ask alex